posthog/hogql_queries/insights/trends/aggregation_operations.py

```diff
@@ -232,39 +282,82 @@ def _inner_select_query(
                 GROUP BY d.timestamp
                 ORDER BY d.timestamp
             """,
-            placeholders={
-                **self.query_date_range.to_placeholders(),
-                **self._interval_placeholders(),
-                "cross_join_select_query": cross_join_select_query,
-            },
+                placeholders={
+                    **self.query_date_range.to_placeholders(),
+                    **self._interval_placeholders(),
+                    "cross_join_select_query": cross_join_select_query,
+                },
+            ),
         )
 
+        if self.should_aggregate_values:
+            query.select = [ast.Field(chain=["d", "timestamp"]), ast.Field(chain=["actor_id"])]
+            query.group_by.append(ast.Field(chain=["actor_id"]))
+
+        return query
+
     def _events_query(
         self, events_where_clause: ast.Expr, sample_value: ast.RatioExpr
     ) -> ast.SelectQuery | ast.SelectUnionQuery:
-        if self._is_count_per_actor_variant():
-            return parse_select(
+        date_filters = [
+            parse_expr(
+                "timestamp >= {date_from} - {inclusive_lookback}",
+                placeholders={
+                    **self.query_date_range.to_placeholders(),
+                    **self._interval_placeholders(),
+                },
+            ),
+            parse_expr(
+                "timestamp <= {date_to}",
+                placeholders={
+                    **self.query_date_range.to_placeholders(),
+                    **self._interval_placeholders(),
+                },
+            ),
```

Comment:
Both ends of the timestamp are `>=` and `<=`. Usually I'd expect to see on bounded, and one unbounded. Bug or on purpose?

Reply:
Great question, lemme investigate

Reply:
Checked, it's intentional